src/commands/trace/trace.ts

```diff
@@ -0,0 +1,144 @@
+import chalk from 'chalk'
+import {spawnSync} from 'child_process'
+import {Command} from 'clipanion'
+import crypto from 'crypto'
+import os from 'os'
+import {parseTags} from '../../helpers/tags'
+import {apiConstructor} from './api'
+import {APIHelper, CIRCLECI, Provider, SUPPORTED_PROVIDERS} from './interfaces'
+
+// We use 127 as exit code for invalid commands since that is what *sh terminals
+const badCommandExitCode = 127
+
+export class TraceCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Trace a command with a custom span and report it to Datadog.',
+    details: `
+            This command wraps another command, which it will launch, and report a custom span to Datadog.
+            See README for details.
+        `,
+    examples: [
+      ['Trace a command and report to Datadog', 'datadog-ci trace --name "Say Hello" -- echo "Hello World"'],
```

Comment:
maybe add here something like 
```suggestion
      ['Trace a command with name "Say Hello" and report to Datadog', 'datadog-ci trace --name "Say Hello" -- echo "Hello World"'],
```